### PR TITLE
Remove unneeded status checker for fsdp test

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

#17710 introduced a test with a status checker for dead processes. We've seen flakiness in the CI with ranks failing, and my suspicion is that this status checker is buggy. In any case, it shouldn't be needed since we are running the test standalone. Please correct me if I'm wrong, but the tests are passing without this tool, so at the very least this makes the test a bit lighter.

```
Ran	strategies/test_fsdp.py::test_apply_optimizer_in_backward[checkpoint0]
Ran	strategies/test_fsdp.py::test_apply_optimizer_in_backward[checkpoint1]
Ran	strategies/test_fsdp.py::test_apply_optimizer_in_backward[checkpoint2]
Ran	strategies/test_fsdp.py::test_apply_optimizer_in_backward[None]
```

If this doesn't address the flakiness, the next course of action I propose is to move the test to the same folder where we run parity tests with rerun attempts for combatting flakiness.

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

cc @carmocca @justusschock @awaelchli @borda

### CODE REVIEW

File: tests/tests_fabric/strategies/test_fsdp.py
Comment (awaelchli) [inline tests/tests_fabric/strategies/test_fsdp.py:396]: The test requires @RunIf(min_torch="2.0.0") and so the import will work

**Review by Borda — APPROVED** [review]

**Review by robieta — APPROVED** [review]
It's not obvious to me that this will fix the flakiness, but you're welcome to try. (And as you say, `standalone` should be sufficient.)

**Review by carmocca — APPROVED** [review]

**Comment by awaelchli** [comment]
I'll hold off merging this, we want to investigate. But as discussed I will skip the test temporarily in another PR that is ready to merge (once GH outage is over).

**Pull request closed without merging**
